js/index.js

```diff
@@ -77,7 +95,87 @@ function renderRain(rainPercentage) {
+
+    element.innerText = temp;
+}
+
+
 function convertKelvinToCelsius(tempInKelvin) {
     const diffrence = 273.15;
     return tempInKelvin - diffrence;
-}
\ No newline at end of file
+}
+
+const getWeekFromToday = (date) => {
+    const weekFromToday = [];
+    weekFromToday.push(...days.slice(date.getDay(), days.length),...days.slice(0, date.getDay()) );
+    return weekFromToday.map((day, index)=> {
+        const dayDate = new Date(date.getTime());
+        dayDate.setDate(date.getDate() + index);
+
+        return {
+            day,
+            index,
+            date: dayDate,
+            month: monthNames[date.getMonth()]
+        }
+    });
+}
+
+const renderForecastData = async () => {
+    const forecastData = await getForecast("Warsaw");
+    const weekFromToday = getWeekFromToday(new Date());
+    const daysHTML = weekFromToday.map((dataObject, index) => {
+        if (index === 0) {
+            return `
+<div class="today forecast">
+    <div class="forecast-header">
+    <div class="day">${dataObject.day}</div>
+    <div class="date">${dataObject.date.getDate()} ${dataObject.month}</div>
+    <button class="refresh-button" id="refresh-weather">
+    <i class="fa fa-refresh" aria-hidden="true"></i>
+    </button>
+    </div> <!-- .forecast-header -->
+    <div class="forecast-content">
+        <div class="location">Warsaw</div>
+        <div class="degree">
+            <div class="num"><span id="current-temp">NA</span><sup>o</sup>C</div>
+            <div class="forecast-icon">
+                <img src="images/icons/icon-1.svg" alt="" width=90>
+            </div>
+        </div>
+        <span><img src="images/icon-umberella.png" alt=""><span id="current-rain"
+        style="margin-right: 0;">NA</span></span>
+        <span><img src="images/icon-wind.png" alt=""><span id="wind" style="margin-right: 5px;">NA</span>km/h</span>
+        <span><img src="images/icon-compass.png" alt="">East</span>
+    </div>
+</div>`
+        }
+
+        const forecastIndex = (index - 1) * 8 + 4;
+        let forecastNight = forecastData.list[forecastIndex - 4];
+        let forecastDay = forecastData.list[forecastIndex];
+        if (forecastDay === undefined) {
+            forecastDay = forecastData.list[36];
+            forecastNight = forecastData.list[32];
+        }
+
+
+        return `<div class="forecast">
+            <div class="forecast-header">
+            <div class="day">${dataObject.day}</div>
+            </div> <!-- .forecast-header -->
+            <div class="forecast-content">
+            <div class="forecast-icon">
+            <img src="images/icons/icon-3.svg" alt="" width=48>
+            </div>
+            <div class="degree">${convertKelvinToCelsius(forecastDay.main.temp).toFixed(2)}<sup>o</sup>C</div>
+            <small>${convertKelvinToCelsius(forecastNight.main.temp).toFixed(2)}<sup>o</sup></small>
+              </div>
+        </div>`
+    }).join("");
```

Comment:
Nie podoba mi się renderowanie całego kodu HTML. Myślę, że lepiej byłoby zmieniać wybrane wartości w kodzie strony. Rozwiązało by to również problem z `await` z początku kodu.